sycl/include/CL/sycl/atomic.hpp

```diff
@@ -21,8 +21,9 @@
 #include <type_traits>
 
 #define __SYCL_STATIC_ASSERT_NOT_FLOAT(T)                                      \
-  static_assert(!std::is_same<T, float>::value,                                \
-                "SYCL atomic function not available for float type")
+  static_assert(                                                               \
+      !std::is_same<T, float>::value && !std::is_same<T, double>::value,       \
+      "SYCL atomic function not available for float and double types")
```

Comment:
We shouldn't add `double` support to the legacy `atomic` types.  These were deprecated in SYCL 2020 (https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:atom-types-depr), and we shouldn't add new functionality to deprecated features.

Reply:
@Pennycook Thank you for the answer. I've reverted the commit that adds `double` to the `atomic.hpp` header.